src/components/PageDefinitions.js

```diff
@@ -87,20 +88,34 @@ class PageDefinitions extends AbstractPageDefinitions {
     }
   }
 
+  // Get an array of definitions asynchronous, split them into 100 chunks and alert the user when they're all done
+  getDefinitionsAndNotify(definitions, message) {
+    const { dispatch, token } = this.props
+    const chunks = chunk(definitions, 100)
+    Promise.all(chunks.map(chunk => dispatch(getDefinitionsAction(token, chunk))))
```

Comment:
Might consider using `throat(10)` or something here to not abuse the service. If the UI is showing 2000 definitions then refresh would do 20 parallel 100 element requests. Just be friendly.

Reply:
Sounds fair. I've done that